src/traces/histogram/calc.js

```diff
@@ -435,6 +435,7 @@ function getConnectedHistograms(gd, trace) {
     for(var i = 0; i < fullData.length; i++) {
         var tracei = fullData[i];
         if(tracei.type === 'histogram' &&
+            tracei.visible === true &&
```

Comment:
To match the condition [above](https://github.com/plotly/plotly.js/pull/2724/files#diff-e02f19bac6aa5dbda7a48a07b3ad45fdR27) filtering out invisible traces. Interestingly the initial draw (if you just set `visible: 'legendonly'` on the first trace as of `newPlot`) works OK, though I suspect that in certain cases this will lead us to choose the wrong autobins... this fixes that potential problem as well.